hdfs3/tests/test_hdfs3.py

```diff
@@ -159,6 +159,15 @@ def test_write_blocksize(hdfs):
         hdfs.open(a, 'rb', block_size=123)
 
 
+def test_write_vbig(hdfs):
+    with hdfs.open(a, 'wb') as f:
+        f.write(b' ' * 2**31)
```

Comment:
It was at this kind of scale that I noticed the problem - it's not related to blocksize per se. 
I can think of the following possibilities:
- you provide the data length to libdhfs3 as a 32-bit int, so data lengths this big or above will over flow
- a simple time-out in copying the data from python to whatever buffering is required in the write.